extensions/wikia/AdEngine/js/tracking/adInfoTracker.js

```diff
@@ -0,0 +1,93 @@
+/*global define, require*/
+define('ext.wikia.adEngine.tracking.adInfoTracker',  [
+	'ext.wikia.adEngine.adTracker',
+	'ext.wikia.adEngine.slot.service.slotRegistry',
+	'wikia.browserDetect',
+	'wikia.log',
+	'wikia.window',
+	require.optional('ext.wikia.adEngine.ml.rabbit')
+], function (adTracker, slotRegistry, browserDetect, log, win, rabbit) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.tracking.adInfoTracker';
+
+	function prepareData(slotName, pageParams, slotParams, creative, bidders) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has a complexity of 34 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has 5 parameters which is greater than 4 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-js-rules%3AAmdExcludedFunctionParameterLimit)